torch/distributions/lkj_cholesky.py

```diff
@@ -0,0 +1,125 @@
+"""
+This closely follows the implementation in NumPyro (https://github.com/pyro-ppl/numpyro).
+
+Original copyright notice:
+
+# Copyright: Contributors to the Pyro project.
+# SPDX-License-Identifier: Apache-2.0
+"""
+
+import math
+
+import torch
+from torch.distributions import constraints, Beta
+from torch.distributions.distribution import Distribution
+from torch.distributions.utils import broadcast_all
+
+
+class LKJCholesky(Distribution):
```

Comment:
The name seems weird to me, but maybe it's ok since we already have `LowRankMultivariateNormal`.

Reply:
The `LKJ` or `LKJCorr` distribution in the paper samples from the space of positive definite correlation matrices, but this one samples the Cholesky factor of correlation matrices. The name is to help clarify this. TFP calls this `CholeskLKJ` and Stan calls this `lkj_corr_cholesky`. I'm happy to change it to something else if you have any suggestions.

Reply:
Thanks for the explanation, in lieu of this we could retain the name.